examples/versioning.ipynb

```diff
@@ -0,0 +1,670 @@
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from openassetio.access import ResolveAccess\n",
+    "from openassetio_mediacreation.traits.content import LocatableContentTrait\n",
+    "from openassetio_mediacreation.traits.identity import DisplayNameTrait\n",
+    "from openassetio_mediacreation.traits.lifecycle import VersionTrait\n",
+    "\n",
+    "# We can re-use the list of references to batch-resolve severeal traits for \n",
+    "# all versions at once. This can be useful for buliding menus/pickers/etc...\n",
+    "# Batching helps mitigate the cost of expensive database lookups that incur\n",
+    "# a fixed per-call cost due to network latency\n",
+    "all_versions_entity_data = manager.resolve(\n",
+    "    versions_refs, \n",
+    "    {LocatableContentTrait.kId, DisplayNameTrait.kId, VersionTrait.kId}, \n",
+    "    ResolveAccess.kRead, context)\n",
+    "\n",
+    "# Display some rudimentary information about each version\n",
+    "for entity_data in all_versions_entity_data:\n",
+    "    name = DisplayNameTrait(entity_data).getName()\n",
+    "    version_tag = VersionTrait(entity_data).getStableTag()\n",
+    "    print(f\"{name} (Version '{version_tag}')\")\n",
+    "    helpers.display_image(LocatableContentTrait(entity_data).getLocation())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a05b7b25-6414-4679-8dfa-d34c0a0d24b5",
+   "metadata": {
+    "editable": true,
+    "jp-MarkdownHeadingCollapsed": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Querying a specific version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "62c0dea8-42ed-47a0-b62a-fffe552ddf05",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "We can also set trait properties when we make a relationship query, and this will act as a filter. we're going to use the `getWithRelationship` method with the unversioned entity reference, to query the reference for a specific version of the entity - v1 - using a specific version tag (perhaps entered by the user)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fa2020ad-f6bb-458d-8ab1-8dd9b93dc1ce",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from openassetio.access import RelationsAccess\n",
+    "\n",
+    "from openassetio_mediacreation.specifications.lifecycle import EntityVersionsRelationshipSpecification\n",
+    "\n",
+    "# Define the relationship we wish to query, and the specific\n",
+    "# tag we want to look up.\n",
+    "relationship_spec = EntityVersionsRelationshipSpecification.create()\n",
+    "relationship_spec.versionTrait().setSpecifiedTag(\"1\")\n",
```

Comment:
[consider] Is it worth a note that `setSpecifiedTag` must follow the same format as (and likely have come from some list populated using) `version_trait.getSpecifiedTag()`?

Reply:
Added a note [here](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/59/commits/0fec8b93acfb2956f6939e92ddd975a9f8af0142)

Reply:
:+1: though typo "perviously"